src/npc.cpp

```diff
@@ -294,6 +302,19 @@ void Npc::onCreatureMove(Creature* creature, const Tile* newTile, const Position
 		if (m_npcEventHandler) {
 			m_npcEventHandler->onCreatureMove(creature, oldPos, newPos);
 		}
+
+		if (creature != this) {
+			Player* player = creature->getPlayer();
+
+			// if player is now in range, add to spectators list, otherwise erase
+			if (player->canSee(_position)) {
```

Comment:
I think any creature will trigger the function. I'm sure at least other NPCs trigger the onCreatureMove callback.